### PR TITLE
feat!: update to work with w3up api

### LINKED ISSUE #87: Upgrade to work with w3up
This issue is to document how to use add-to-web3 with a w3up space. https://blog.web3.storage/posts/the-data-layer-is-here-with-the-new-web3-storage

I imagine I might need to pass a UCAN delegation to it and maybe also pass key material from a github secret that the ucan delegates to and that add-to-web3 can use to sign invocations of upload/add
```[tasklist]
### Tasks
- [ ] https://github.com/web3-storage/add-to-web3/pull/89
```

### PR DESCRIPTION
**Pull request by olizilla**
Rewrite of the add-to-web3 github action to work with the new w3up web3.storage api.

[`w3cli`](https://github.com/web3-storage/w3cli) does everything we need to upload files to web3.storage, so this action now provides a very thin wrapper around that cli in the form of a composite action.

## Breaking changes

Auth is based on [UCAN](https://web3.storage/docs/concepts/ucans-and-web3storage/)s now. Use w3cli to create a signing key for this action to use, and create a ucan that delegates upload abilities to that key.

- input: `web3_token` has been removed, as token based auth as been removed.
  - set `secret_key` to a base64 encoded siging key you create via `w3 key create`.
  - set `proof` to the a base64 CAR encoded UCAN delegating capabilities to the secret_key's DID.
- input: `wrap_with_directory` has been removed. w3cli does not wrap directories in directories by default now. Only single files get wrapped to preserve their filename.
  - set `wrap: false` to disable the wrapping of single files.
- input: `web3_api` has been removed. Targeting other services is not currently supported. Open an issue if you need that.
- input: `include_hidden` has been renamed to `hidden` to match the w3cli flag name

Fixes https://github.com/web3-storage/add-to-web3/issues/87

Needs https://github.com/web3-storage/w3cli/pull/158

License: MIT

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -4,30 +4,65 @@ author: olizilla
...
+      run: jq --exit-status --raw-output '.root."/"' ./w3_up_output.json
+      shell: bash
+
+    - id: get_cid
+      run: echo "::set-output name=cid::$(jq --exit-status --raw-output '.root."/"' ./w3_up_output.json)"
+      shell: bash
+
+    - id: get_url
+      run: echo "::set-output name=url::https://${{ steps.get_cid.outputs.cid }}.ipfs.w3s.link"
+      shell: bash
```
Comment (olizilla) [inline action.yml:68]: deprecation warning
> Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -7,27 +7,14 @@ on:
...
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - uses: ./
       with:
-        web3_api: 'https://api-staging.web3.storage'
-        web3_token: ${{ secrets.WEB3_STORAGE_TOKEN }}
+        proof: ${{ secrets.W3_PROOF }}
+        secret_key: ${{ secrets.W3_PRINCIPAL }}
```
Comment (vasco-santos) [inline .github/workflows/test.yml:17]: nit: consider using same names here instead of secret key and principal?

File: README.md [hunk 3]
```diff
@@ -25,24 +27,20 @@ with:
...
+Create one using `w3 key create`. See: https://github.com/web3-storage/w3cli#w3_principal
 
-Should the `path_to_add` be wrapped in a diretory when creating the IPFS DAG. For most folks using this action the default of `false` is fine. 
+### `proof`
 
-This is the opposite of the default that web3.storage uses, as this action is commonly used to add a directory that contains a static website to IPFS. In that case you want the path_to_add to become the root cid so you can host your site at `https://<cid>.ipfs.dweb.link` rather than `https://<cid>.ipfs.dweb.link/<path_to_add>`.
+**Required** A base64 encoded UCAN delegating capabilities the signing key above. 
 
-If you do want to capture the `path_to_add` path itself in the IPFS DAG then you want to set `wrap_with_directory:true`.
+Create using `w3 delegation create --base64`
```
Comment (vasco-santos) [inline README.md:40]: should we mention in the docs that delegation can be narrowed to a set of capabilities and they should make it only for their needs pointing to the docs https://github.com/web3-storage/w3cli?tab=readme-ov-file#w3-delegation-create-audience-did ?

**Review by vasco-santos — APPROVED** [review]
LGTM! 🎉  small nits in review comments

File: action.yml [hunk 4]
```diff
@@ -4,30 +4,65 @@ author: olizilla
...
   icon: 'box'
   color: 'white'
+
 inputs:
   path_to_add:
     description: 'Directory path to add to IPFS'
     required: true
+
+  secret_key:
+    description: 'The base64 key to use to sign ucan invocations. Create one using `w3 key create`. See: https://github.com/web3-storage/w3cli#w3_principal'
```
Comment (alanshaw) [inline action.yml:14]:
```suggestion
    description: 'The base64 encoded key to use to sign UCAN invocations. Create one using `w3 key create`. See: https://github.com/web3-storage/w3cli#w3_principal'
```

File: action.yml [hunk 5]
```diff
@@ -4,30 +4,65 @@ author: olizilla
...
+    required: true
+
+  proof:
+    description: 'A base64 encoded UCAN delegating capabilities the signing key above. Create using `w3 delegation create --base64`'
```
Comment (alanshaw) [inline action.yml:18]:
```suggestion
    description: 'A base64 encoded UCAN delegating capabilities the secret signing key above. Create using `w3 delegation create --base64`'
```

File: README.md (hunk 3 above)
Comment (alanshaw) [reply README.md:40]: yeah it is good practice for us to recommend restricting caps.

File: README.md [hunk 6]
```diff
@@ -1,22 +1,24 @@
-<h1 align="center">⁂<br/>web3.storage</h1>
-<p align="center">Add a directory to web3.storage from an Action, and output it's IPFS Content ID.</p>
+# add-to-web3
 
-! v3 of this action will work with the new w3up api once https://github.com/web3-storage/add-to-web3/issues/87 lands.
+Upload files to web3.storage from a Github Action, and output it's IPFS Content ID.
+
+_A composite github action. It's [just yaml](./action.yml) calling [w3cli](https://github.com/web3-storage/w3cli_
```
Comment (alanshaw) [inline README.md:5]:
```suggestion
_A composite github action. It's [just yaml](./action.yml) calling [w3cli](https://github.com/web3-storage/w3cli)_
```

**Review by alanshaw — APPROVED** [review]
LGTM, I have left some feedback for your considerations.

**Review by Gozala — APPROVED** [review]
❤️

**Review by Fool256 — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,62 +1,94 @@
-<h1 align="center">⁂<br/>web3.storage</h1>
-<p align="center">Add a directory to web3.storage from an Action, and output it's IPFS Content ID.</p>
+# add-to-web3
 
-! v3 of this action will work with the new w3up api once https://github.com/web3-storage/add-to-web3/issues/87 lands.
+Upload files to [web3.storage](https://web3.storage) from a Github Action, and output the IPFS Content ID.
```
Comment (Fool256) [inline README.md:4]: Also if Web 3 wants to be accurate I'll will choose keccake256 and Keccake512

**Review by Fool256 — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (Fool256) [reply .github/workflows/test.yml:34]: Felix=babel time

**Review by Fool256 — APPROVED** [review]

**Review by Fool256 — APPROVED** [review]

**Pull request merged**